### ISSUE TITLE
[Bug]: Crash with FTB Quests Custom Quest

### ISSUE THREAD

**Issue by Crepper710** [FTBTeam/FTB-Mods-Issues#407]
### Mod

FTB Quests

### Mod version

ftb-quests-forge-1802.3.8-build.101

### Forge / Fabric version

Forge 1.18.2 40.1.68

### What issue are you having?

I tried to code a custom quest for FTB Quest using KubeJS (like in the example at: https://mods.latvian.dev/books/kubejs/page/ftb-quests-integration)

the code i put in kubejs/startup_scripts/script.js or kubejs/server_scripts/script.js (neither worked):

```
onEvent('ftbquests.custom_task.620618336B00D868', event => {
    console.log('Custom task!')
    event.setCheckTimer(20);
    event.check = (task, player) => {
        if (player.isOp()) {
            task.progress = 1
        }
    }
})
```

And every time i load up a single player world, my client crashes as soon as the world is loaded.

### Crashlogs

crash report: https://paste.feed-the-beast.com/view/03d66c0f
latest.log: https://paste.feed-the-beast.com/view/76b445db

### Steps to reproduce

1. I put the above mentioned code in kubejs/startup_scripts/script.js or kubejs/server_scripts/script.js
2. I join a world
3. Game crashes

### Anything else to note?

I already created an issue at the github page of KubeJS, since i am not sure which mod is doing something wrong and KubeJS and the classes crashing in FTB Quests are by the same coder. ([link](https://github.com/KubeJS-Mods/KubeJS/issues/446))

**Comment by Raidobw2** [comment]
ServerPlayer should probably be PlayerSelector in
https://github.com/FTBTeam/FTB-Quests/blob/9f748aa674afa1b747948d16690bc3e82fe07d30/common/src/main/java/dev/ftb/mods/ftbquests/integration/kubejs/CheckWrapper.java#L18

because of https://github.com/KubeJS-Mods/KubeJS/blob/605ad3fcd642eae66517edefd9cdf8215d91c547/common/src/main/java/dev/latvian/mods/kubejs/server/ServerJS.java#L201

There might be other stuff to check, though haven't dug more than the error showed

Basically Crepper710, it's not currently possible to use that code with this version of FTB Quests, FTB Quests will need to be updated

**Comment by Crepper710** [comment]
I looked through the branches and decompiled the Mod.

It looks like there is already a correct implementation:
https://github.com/FTBTeam/FTB-Quests/blob/1.18/main/common/src/main/java/dev/ftb/mods/ftbquests/integration/kubejs/CheckWrapper.java#L19

but the Mod on CurseForge uses:
https://github.com/FTBTeam/FTB-Quests/blob/main/common/src/main/java/dev/ftb/mods/ftbquests/integration/kubejs/CheckWrapper.java#L18

**Comment by Crepper710** [comment]
I looked through the branches and decompiled the Mod.

It looks like there is already a correct implementation:
https://github.com/FTBTeam/FTB-Quests/blob/1.18/main/common/src/main/java/dev/ftb/mods/ftbquests/integration/kubejs/CheckWrapper.java#L19

but the Mod on CurseForge uses:
https://github.com/FTBTeam/FTB-Quests/blob/main/common/src/main/java/dev/ftb/mods/ftbquests/integration/kubejs/CheckWrapper.java#L18


I also found a workaround, just use an older version of KubeJS. (1802.5.3-build.461 and before)

**Comment by desht** [comment]
Next release of FTB Quests will work correctly with up-to-date KubeJS builds.

**Comment by desht** [comment]
Implemented for 1.18 & 1.19
